engine/class_modules/sc_death_knight.cpp

```diff
@@ -1659,6 +1660,30 @@ struct death_knight_pet_t : public pet_t
 
     return m;
   }
+
+  double composite_melee_haste() const override
+  {
+    double haste = pet_t::composite_melee_haste();
+
+    if (o() -> legendary.phearomones -> ok())
+    {
+      haste *= 1.0 / (1.0 + o() -> buffs.death_turf -> check_value());
+    }
+
+    return haste;
+  }
+
+  double composite_spell_haste() const override
+  {
+    double haste = pet_t::composite_spell_haste();
+
+    if (o() -> legendary.phearomones -> ok())
+    {
+      haste *= 1.0 / (1.0 + o() -> buffs.death_turf -> check_value());
+    }
+
+    return haste;
+  }
```

Comment:
You may want to gate that behind a owner -> bugs check and add comments about the double dipping.